infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -88,6 +90,9 @@ public class LogReplicationMetadataManager {
     @Setter
     private long topologyConfigId;
 
+    @Getter
+    private final AtomicBoolean isLeader;
```

Comment:
I see that some of the methods in the file are guarded by isLeader check, while other methods are still open for non-leader nodes. For example. There is an addSession method which is open for non-leaders. Is it intended?

Another question: Are we instantiating LogReplicationMetadataManager for non-leader nodes as well? If yes, can we avoid it? That way, we don't need to guard these methods with isLeader check.

Reply:
yes its intended.
(1) the reason being that its good to have in-memory sessions maintained...so a topology notification is not missed by non-leader nodes. This way if the leadership changes, the new leader has the latest topology in memory.
(2) Also, `addSession()` is open only for the leader node..
(3) the other methods which do not check `isLeader`, are guaranteed to be called only once the leadership is acquired... they come in the flow that happens only for the leader node.